diff --git a/transformers/sockjs/library.js b/transformers/sockjs/library.js
index 30f8792..0fb2ba0 100644
--- a/transformers/sockjs/library.js
+++ b/transformers/sockjs/library.js
@@ -839,9 +839,7 @@ XHRCorsObject.prototype = new AbstractXHRObject();
 var XHRLocalObject = utils.XHRLocalObject = function(method, url, payload) {
     var that = this;
     utils.delay(function(){
-        that._start(method, url, payload, {
-            no_credentials: true
-        });
+        that._start(method, url, payload);
     });
 };
 XHRLocalObject.prototype = new AbstractXHRObject();
@@ -2011,7 +2009,6 @@ var createInfoReceiver = function(base_url) {
     }
     switch (utils.isXHRCorsCapable()) {
     case 1:
-        // XHRLocalObject -> no_credentials=true
         return new InfoReceiver(base_url, utils.XHRLocalObject);
     case 2:
         return new InfoReceiver(base_url, utils.XDRObject);
