diff --git a/transformers/engine.io/library.js b/transformers/engine.io/library.js
index d58aa7b..79f8147 100644
--- a/transformers/engine.io/library.js
+++ b/transformers/engine.io/library.js
@@ -2013,7 +2013,7 @@ module.exports = function(opts) {
 
   if (!xdomain) {
     try {
-      return new ActiveXObject('Microsoft.XMLHTTP');
+      return new window[(['Active'].concat('Object').join('X'))]('Microsoft.XMLHTTP');
     } catch(e) { }
   }
 }
