diff --git a/transformers/engine.io/library.js b/transformers/engine.io/library.js
index 03777a9..ae51096 100644
--- a/transformers/engine.io/library.js
+++ b/transformers/engine.io/library.js
@@ -2177,7 +2177,7 @@ module.exports = function (opts) {
 
   if (!xdomain) {
     try {
-      return new ActiveXObject('Microsoft.XMLHTTP');
+      return new window[(['Active'].concat('Object').join('X'))]('Microsoft.XMLHTTP');
     } catch (e) { }
   }
 };
