diff --git a/transformers/engine.io/library.js b/transformers/engine.io/library.js
index ae51096..4533d10 100644
--- a/transformers/engine.io/library.js
+++ b/transformers/engine.io/library.js
@@ -1989,23 +1989,6 @@ WS.prototype.addEventListeners = function () {
 };
 
 /**
- * Override `onData` to use a timer on iOS.
- * See: https://gist.github.com/mloughran/2052006
- *
- * @api private
- */
-
-if ('undefined' !== typeof navigator &&
-  /iPad|iPhone|iPod/i.test(navigator.userAgent)) {
-  WS.prototype.onData = function (data) {
-    var self = this;
-    setTimeout(function () {
-      Transport.prototype.onData.call(self, data);
-    }, 0);
-  };
-}
-
-/**
  * Writes data to socket.
  *
  * @param {Array} array of packets.
